tests/network/validations.go

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"fmt"
+
+	k8sv1 "k8s.io/api/core/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+)
+
+// ValidateVMIandPodIPMatch Checks that the vmi pod and vmi scheme have matching Ip/Ips fields for primary interface
+func ValidateVMIandPodIPMatch(vmi *v1.VirtualMachineInstance, vmiPod *k8sv1.Pod) error {
+	if vmi.Status.Interfaces[0].IP != vmiPod.Status.PodIP {
```

Comment:
i would suggest to add PodIPs[0] == PodIP
(also suggested by Dan)
wdyt ?

at least to have it in the tests

Reply:
Who is "Dan"? You can ask "him" to comment here or you can give your arguments for it.

In any case, to the point, this helper checks that the IP/s are identical between VMI/s and Pod/s, not that they correct (including that the first in the list equals the old single one).

Reply:
I just wanted to give credit that it was discussed with Dan Kenigsberg.

Anyhow, from my understanding, the reason is that the tests should assert validations that are considered prerequisite by the core code and to add it to an already existing test.

I thought this is a simple assert and can be added to this function even if it runs in couple tests (since its simple check), 
or at least add it to one place outside of this function.
So we have at least one assert the core logic prerequisite does not break.

Reply:
The `PodIPs[0] == PodIP` is a virt-launcher pod validation so we either create new validation function for that ,something like `ValidateVirtLauncherPodIPs`, or if we add this to this function we remove te Matching to generalize it a little.

Reply:
PodIPs[0] == PodIP is now also a fact on the VMI, there i meant to verify it.
didnt understand the end of the sentence "or if we add ..." can you rephrase please?

Reply:
Ok let's just verify it

Reply:
@EdDev is it ok that i will add to this function as Quique also suggested now ?

Reply:
I do not think it is worth a function... it is a simple assert with almost zero logic.
I also do not think we need to test it in every test, one simple test is enough.

Reply:
please see the EDIT above, i think we dont even need it anymore, do we ?